src/pgduckdb_types.cpp

```diff
@@ -173,24 +148,209 @@ ConvertNumeric(T value, idx_t scale) {
 		fractional_digits[i] = uint16_t(fractional_part % NBASE);
 		fractional_part /= NBASE;
 	}
-
 	ndigits = integral_ndigits + fractional_ndigits;
-
 	result.buf = (NumericDigit *)palloc(ndigits * sizeof(NumericDigit));
 	result.digits = result.buf;
 	auto &digits = result.digits;
-
 	idx_t digits_idx = 0;
 	for (idx_t i = integral_ndigits; i > 0; i--) {
 		digits[digits_idx++] = integral_digits[i - 1];
 	}
 	for (idx_t i = fractional_ndigits; i > 0; i--) {
 		digits[digits_idx++] = fractional_digits[i - 1];
 	}
-
 	return result;
 }
 
+static Datum
+ConvertNumericDatum(const duckdb::Value &value) {
+	if (value.type().id() == duckdb::LogicalTypeId::DOUBLE) {
+		return ConvertDoubleDatum(value);
+	}
+	NumericVar numeric_var;
+	D_ASSERT(value.type().id() == duckdb::LogicalTypeId::DECIMAL ||
+	         value.type().id() == duckdb::LogicalTypeId::HUGEINT ||
+	         value.type().id() == duckdb::LogicalTypeId::UBIGINT);
+	auto physical_type = value.type().InternalType();
+	const bool is_decimal = value.type().id() == duckdb::LogicalTypeId::DECIMAL;
+	uint8_t scale = is_decimal ? duckdb::DecimalType::GetScale(value.type()) : 0;
+
+	switch (physical_type) {
+	case duckdb::PhysicalType::INT16: {
+		numeric_var = ConvertNumeric<int16_t>(value.GetValueUnsafe<int16_t>(), scale);
+		break;
+	}
+	case duckdb::PhysicalType::INT32: {
+		numeric_var = ConvertNumeric<int32_t>(value.GetValueUnsafe<int32_t>(), scale);
+		break;
+	}
+	case duckdb::PhysicalType::INT64: {
+		numeric_var = ConvertNumeric<int64_t>(value.GetValueUnsafe<int64_t>(), scale);
+		break;
+	}
+	case duckdb::PhysicalType::UINT64: {
```

Comment:
I know you didn't write this code, just moved it around. But reading this now it makes me wonder. Why is this the only unsigned integer type that we check for the physical type? @mkaruza do you know?